src/apps/activity_assignments/domain/assignments.py

```diff
@@ -56,3 +61,34 @@ class ActivityAssignmentWithSubject(PublicModel):
 class ActivitiesAssignmentsWithSubjects(PublicModel):
     applet_id: UUID
     assignments: list[ActivityAssignmentWithSubject]
+
+
+class ActivityAssignmentDelete(BaseModel):
+    activity_id: UUID | None
+    activity_flow_id: UUID | None
+    respondent_subject_id: UUID
+    target_subject_id: UUID
+
+    @root_validator
+    def validate_assignments(cls, values):
+        # Validate that exactly one of activity_id or activity_flow_id is provided
+        if not values.get("activity_id") and not values.get("activity_flow_id"):
+            raise ActivityAssignmentNotActivityAndFlowError()
+
+        # Validate that respondent_subject_id is provided
+        if not values.get("respondent_subject_id"):
+            raise ActivityAssignmentMissingRespondentError()
+
+        # Validate that target_subject_id is provided
+        if not values.get("target_subject_id"):
+            raise ActivityAssignmentMissingTargetError()
```

Comment:
I don't think you need to add extra validations on the `respondent_subject_id` and `target_subject_id` properties – the strict typing above (`None` not being allowed) already causes Pydantic to validate these properties correctly if either of these are omitted:

![CleanShot 2024-08-27 at 18 19 13@2x](https://github.com/user-attachments/assets/22dd8fcd-e7eb-4db7-8224-9f690f31af45)

It's not wrong to add your version, it's just a bit redundant. If it's for improved readability and you'd like to keep them, then please also do the same in `ActivityAssignmentCreate` so that there's consistency. But I'd probably just omit them, as elsewhere we generally leave it up to Pydantic to perform these checks automatically. My understanding is that the purpose of these additional validators is for handling more nuanced cases that can't be described with strict typing.

Reply:
I agreed with @farmerpaul, we don't need these extra validations, pydantic already do that